analysis_driver/segmentation.py

```diff
@@ -17,31 +18,25 @@ def serialize(self, x):
 
 class BasicStage(luigi.Task, AppLogger):
     __stagename__ = None
-    previous_stages = []
     exit_status = None
 
+    previous_stages = EGCGParameter(default=[])
     dataset = EGCGParameter()
 
+    def output(self):
```

Comment:
Lock file to allow a BasicStage to track itself without needing a stage in the Rest API. If we're sure we'd like to have this, we'll need to touch this file when exit_status == 0.